CodeEdit/NavigatorSidebar/NavigatorSidebar.swift

```diff
@@ -9,39 +9,43 @@ import SwiftUI
 import WorkspaceClient
 
 struct NavigatorSidebar: View {
-    @ObservedObject var workspace: WorkspaceDocument
-    var windowController: NSWindowController
+	@ObservedObject var workspace: WorkspaceDocument
+	var windowController: NSWindowController
 	@State private var selection: Int = 0
 
-    var body: some View {
-        ZStack {
-            switch selection {
-            case 0:
-                List {
-                    Section(header: Text(workspace.fileURL?.lastPathComponent ?? "Unknown")) {
-                        ForEach(
-                            workspace.selectionState.fileItems.sortItems(foldersOnTop: workspace.sortFoldersOnTop)
-                        ) { item in // Instead of OutlineGroup
-                            NavigatorSidebarItem(
-                                item: item,
-                                workspace: workspace,
-                                windowController: windowController
-                            )
-                        }
-                    }
-                }
-            case 2:
-                SidebarSearch(state: workspace.searchState ?? .init(workspace))
-            default:
+	var body: some View {
+		ZStack {
+			switch selection {
+			case 0:
+				ScrollView {
```

Comment:
Why not using List here. Then you can use selection to implement the sidebar item highlight without comparing item's id with selected id.

Reply:
Because this happens

<img width="242" alt="Screen Shot 2022-03-24 at 18 35 06" src="https://user-images.githubusercontent.com/9460130/159976550-5603574d-2c69-4690-9b24-34f0ecc2f873.png">

Reply:
Also I think this `case 0` part is the real `NavigatorSidebar`. You can extract it as a component and rename this file  `Sidebar` or something.

Reply:
`LeadingSidebar` comes to my mind

Reply:
> Because this happens
> 
> <img alt="Screen Shot 2022-03-24 at 18 35 06" width="242" src="https://user-images.githubusercontent.com/9460130/159976550-5603574d-2c69-4690-9b24-34f0ecc2f873.png">

Actually, Xcode has the outline too.
<img width="313" alt="截屏2022-03-25 上午1 50 28" src="https://user-images.githubusercontent.com/22616933/159979005-7b8deaa3-b9ed-40e0-8b24-7ef93d9d17f2.png">

Reply:
Yes but only on the item itself and not the whole section in a subfolder. But I'll experiment a bit with `List` later this evening.

Reply:
> Yes but only on the item itself and not the whole section in a subfolder

I think that's because You put the whole subfolder in one row.

Reply:
Ok I now switched to `List`

Reply:
But now I have to work on the unfold animation again

Reply:
Got that fixed too!